tests/test_free_energy.py

```diff
@@ -405,3 +414,50 @@ def test_estimate_free_energy_bar_with_energy_overflow():
     assert result_with_nan.dG_err == result_with_inf.dG_err
     np.testing.assert_array_equal(result_with_nan.dG_err_by_component, result_with_inf.dG_err_by_component)
     np.testing.assert_array_equal(result_with_nan.overlap, result_with_inf.overlap)
+
+
+@pytest.mark.parametrize(
+    "n_states,max_delta_states", [(1, 1), (1, None), (1, 2), (2, 1), (3, 6), (6, 3), (6, None), (30, 5)]
+)
+@pytest.mark.parametrize("seed", [2024, 2025])
+def test_compute_potential_matrix(hif2a_ligand_pair_single_topology, n_states: int, max_delta_states: int | None, seed):
+    st, _ = hif2a_ligand_pair_single_topology
+    states = [st.setup_intermediate_state(lam) for lam in np.linspace(0.0, 1.0, n_states)]
+
+    def make_summed_potential(bps: List[BoundPotential]):
```

Comment:
Not for this pr: Would be good to have this utility function somewhere, its very handy.